Src/Particle/AMReX_Particle.H

```diff
@@ -102,6 +102,28 @@ struct ParticleIDWrapper
         r = (sign) ? lval : -lval;
         return r;
     }
+
+    /** Swap the sign of the id.
+     *
+     * Positive ids become negative and negative ids become positive.
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void negate () noexcept
+    {
+        // the leftmost bit is our id sign - we will XOR it with 1 to flip it
+        m_idata ^= static_cast<uint64_t>(1) << 63;
+    }
+
+    /** Check the sign of the id.
+     *
+     * Returns false if the id is positive, otherwise true (invalid particle).
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    bool negative () noexcept
+    {
+        // the leftmost bit is our id sign
+        return m_idata >> 63;
```

Comment:
According to [godbolt](https://godbolt.org/z/K1Yjve6f4), I think I wrote the correct test here ("if the bit is 1, this evaluates to true, so it is negative").

I am confused with the `Long ()` implementation above reading:
```C++
        Long lval = static_cast<Long>(val);  // bc we take -
        r = (sign) ? lval : -lval;
```

Shouldn't that read
```C++
        r = (sign) ? -lval : lval;
```
?

Reply:
For currently unknown reason, we swap the id sign before storing the other 39 bits of it.

Thus, this part needs to read
```suggestion
        return !(m_idata >> 63);
```
for our current convention...

Reply:
The leftmost bit is `0` if the `Long` id is negative and `1` if it's positive, i.e. we store the sign, not it's inverse.

Reply:
Wait, in an int, the leftmost bit is 1 if it is negative

Reply:
I think there is a misunderstanding here in our nomenclature:
- in encoding of signed ints, the sign bit is only non-zero if a number is negative
- here, we currently store if a number has positive sign (the exact opposite)

I think we can simplify this by a lot by changing the convention to the "usual" int sign bit convention (0: positive, 1: negative), but I would not do this in the current PR. Let's chat tomorrow :)